### ISSUE TITLE
Data structure

### ISSUE THREAD

**Issue by RobertBiehl** [BeenKim/MMD-critic#1]
Hi Been,

I liked your presentation a lot and would like to play around with the code on different data sets.
Can you give me a quick hint how to construct the data needed for your program?

Thanks,

Rob

**Comment by rjvak7** [comment]
Hi Robert, the repository contains a representative dataset (usps) that we used to run quantitative performance on. You can start at run_digits.py to see, it should run out of the box.

**Comment by RobertBiehl** [comment]
Yes, I did that, and I also got the image output working. 😊 My question was more targeted towards using other datasets. I presume it takes several steps to preprocess images and create the dataset file to make it work.

-Rob

**Comment by rjvak7** [comment]
Ah ok, great! the usps dataset is in libsvm format with <y> <feature_1:value_1> .... <feature_n:value_n>. You should be able to directly use the code as long as you can get your data to this format. Or, alternatively, you could add your own functions to load data (see data.py) and call prototype/criticism functions from mmd.py . Both methods are equivalent in terms of effort involved and output.
